offline/QA/Jet/photonjetskinematics.cc

```diff
@@ -0,0 +1,184 @@
+//____________________________________________________________________________..
+
+#include "photonjetskinematics.h"
+#include <fun4all/PHTFileServer.h>
+
+#include <fun4all/Fun4AllReturnCodes.h>
+
+#include <phool/PHCompositeNode.h>
+
+#include <TFile.h>
+#include <phool/getClass.h>
+#include "TH1.h"
+#include "TMath.h"
+//#include <jetbase/JetContainer.h>
+
+
+// Tower includes
+#include <calobase/RawCluster.h>
+#include <calobase/RawClusterContainer.h>
+#include <calobase/RawClusterUtility.h>
+#include <calobase/RawTower.h>
+#include <calobase/RawTowerContainer.h>
+#include <calobase/RawTowerGeom.h>
+#include <calobase/RawTowerGeomContainer.h>
+#include <calobase/TowerInfo.h>
+#include <calobase/TowerInfoContainer.h>
+#include <calobase/TowerInfoDefs.h>
+
+// Cluster includes
+#include <calobase/RawCluster.h>
+#include <calobase/RawClusterContainer.h>
+
+
+//____________________________________________________________________________..
+photonjetskinematics::photonjetskinematics(const std::string &outputfilename):
+SubsysReco("photonjetskinematics")
+, outfilename(outputfilename)
+{
+ std::cout << "photonjetskinematics::photonjetskinematics(const std::string &name, const std::string&outputfilename) Calling ctor" << std::endl;
+}
+//____________________________________________________________________________..
+photonjetskinematics::~photonjetskinematics()
+{
+  std::cout << "photonjetskinematics::~photonjetskinematics() Calling dtor" << std::endl;
+}
+
+//____________________________________________________________________________..
+int photonjetskinematics::Init(PHCompositeNode *topNode)
+{
+  std::cout << "photonjetskinematics::Init(PHCompositeNode *topNode) Initializing" << std::endl;
+  //outfile = new TFile(outfilename.c_str(), "RECREATE");
+  PHTFileServer::get().open(outfilename, "RECREATE");
```

Comment:
Instead of sending the histograms to a specific output file, we'll pipe them into the QA hist manager which handles all of that downstream.
```suggestion
manager = QAHistManagerDef::getHistoManager();
if (!manager)
{
  std::cerr << PHWHERE << "PANIC: couldn't grab histogram manager!" << std::endl;
  assert(manager);
}
```